applications/StructuralMechanicsApplication/custom_utilities/EICR.hpp

```diff
@@ -317,58 +320,60 @@ class EICR
-            noalias( Hi ) += eta * prod( Omega, Omega );
+            noalias( Hi ) -= 0.5 * omega;
+            noalias( Hi ) += eta * prod( omega, omega );
 
-            range iRange(index + 3, index + 6);
-            project( H, iRange, iRange ) = Hi;
+            range i_range(index + 3, index + 6);
+            project( H, i_range, i_range ) = Hi;
         }
 
         return H;
     }
 
     /**
     * Computes the Spin derivative of (Axial Vector Jacobian)^T contracted with the nodal moment vector.
-    * The output is a square matrix of size displacements.size() (which is num_nodes * 6).
+    * The output is a square matrix of size rDisplacements.size() (which is num_nodes * 6).
     * Note that 6 Degrees Of Freedom are assumed for each node.
-    * @param displacements the vector of nodal displacements and rotations in the local corotational coordinate system. (assumed size = num_nodes*6)
-    * @param forces the vector of nodal forces and moments in the local corotational coordinate system. (assumed size = num_nodes*6)
-    * @param H the Axial Vector Jacobian Matrix computed with a previous call to EICR::Compute_H(displacements)
+    * @param rDisplacements the vector of nodal displacements and rotations in the local corotational coordinate system. (assumed size = num_nodes*6)
+    * @param rForces the vector of nodal forces and moments in the local corotational coordinate system. (assumed size = num_nodes*6)
+    * @param rH the Axial Vector Jacobian Matrix computed with a previous call to EICR::Compute_H(rDisplacements)
     * @return the L matrix
     */
-    inline static MatrixType Compute_L(const VectorType & displacements, const VectorType & forces, const MatrixType & H)
+    inline static MatrixType Compute_L(const VectorType& rDisplacements,
+                                       const VectorType& rForces,
+                                       const MatrixType& rH)
     {
-        size_t num_dofs = displacements.size();
-        size_t num_nodes = num_dofs / 6;
+        const SizeType num_dofs = rDisplacements.size();
+        const SizeType num_nodes = num_dofs / 6;
 
         MatrixType L(num_dofs, num_dofs, 0.0);
 
-        Vector3Type rotationVector;
-        Vector3Type momentVector;
-        MatrixType Omega(3, 3);
-        MatrixType Omega2(3, 3);
-        MatrixType Li(3, 3);
-        MatrixType LiTemp1(3, 3);
-        MatrixType LiTemp2(3, 3);
+        Vector3Type rotation_vector;
+        Vector3Type moment_vector;
+        BoundedMatrixType3x3 omega(3, 3);
+        BoundedMatrixType3x3 omega_2(3, 3);
+        BoundedMatrixType3x3 Li(3, 3);
+        BoundedMatrixType3x3 LiTemp1(3, 3);
+        BoundedMatrixType3x3 LiTemp2(3, 3);
 
-        for(size_t i = 0; i < num_nodes; i++)
+        for(SizeType i = 0; i < num_nodes; i++)
         {
-            size_t index = i * 6;
-            range iRange(index + 3, index + 6);
-            noalias( rotationVector ) = project( displacements, iRange );

```

Comment:
what does the "project" operator do? i ask since this would be one more thing to be supported by the AMatrix (cc ing Pooyan)

Reply:
I will implement the range support but the interface would be completely different.... I'm not sure if I can mimic the ublas interface on this....